cmd/proxy/proxy.go

```diff
@@ -279,6 +180,62 @@ func (c *Command) initEventBase() {
 	}
 }
 
+func (c *Command) populateConfig() error {
+	if c.flags.config == "" {
+		return nil
+	}
+	b, err := os.ReadFile(c.flags.config)
+	if err != nil {
+		return fmt.Errorf("read config file: %w", err)
+	}
+
+	var config config
+	err = yaml.Unmarshal(b, &config)
+	if err != nil {
+		return fmt.Errorf("parse config file: %w", err)
+	}
+	slog.Debug(fmt.Sprintf("parsed config file, found %d rules", len(config.Rules)))
+
+	env, _ := cel.NewEnv(
+		cel.Variable("request", cel.DynType),
+		cel.Variable("response", cel.DynType),
+	)
+
+	for index, rule := range config.Rules {
+		if !slices.Contains([]string{"include", "exclude"}, strings.ToLower(rule.Then)) {
+			return fmt.Errorf("config: invalid action in rule: %q. Expected either 'include' or 'exclude'", rule.Then)
+		}
+		ast, iss := env.Compile(rule.If)
+		if err = iss.Err(); err != nil {
+			return fmt.Errorf("compile program: %w", err)
+		}
+		if !reflect.DeepEqual(ast.OutputType(), cel.BoolType) {
+			return fmt.Errorf("typecheck program: Got %v, wanted %v result type", ast.OutputType(), cel.BoolType)
+		}
+		program, _ := env.Program(ast)
+		config.Rules[index].Program = program
+	}
+
+	// Test all config rules on a zeroed out request and response as a sanity check
+	for _, rule := range config.Rules {
+		if _, _, err := rule.Program.Eval(map[string]interface{}{
```

Comment:
reuse the `shouldMatch` function instead of having two places with user-facing keywords. this way, the codepath we test at validation time is exactly the same one we use at request time. pass a dummy request and response.

nit: also make this validation a method on the `config`? like, it would be nice to just do `if err := c.validate(); err != nil { ... }` here.